sdk/storage/test/shares/share_file_client_test.cpp

```diff
@@ -0,0 +1,473 @@
+    }
+
+    {
+      // SetProperties works
+      auto client1 = m_fileShareDirectoryClient->GetFileClient(LowercaseRandomString());
+      auto client2 = m_fileShareDirectoryClient->GetFileClient(LowercaseRandomString());
+
+      EXPECT_NO_THROW(client1.Create(1024));
+      EXPECT_NO_THROW(client2.Create(1024));
+      EXPECT_NO_THROW(client1.SetProperties(properties));
+      EXPECT_NO_THROW(client2.SetProperties(properties));
+      auto directoryProperties1 = client1.GetProperties();
+      auto directoryProperties2 = client2.GetProperties();
+      EXPECT_EQ(directoryProperties2->FileCreationTime, directoryProperties1->FileCreationTime);
+      EXPECT_EQ(directoryProperties2->FileLastWriteTime, directoryProperties1->FileLastWriteTime);
+      EXPECT_EQ(directoryProperties2->FileAttributes, directoryProperties1->FileAttributes);
+    }
+  }
+
+  TEST_F(FileShareFileClientTest, HandlesFunctionalityWorks)
+  {
+    auto result = m_fileShareDirectoryClient->ListHandlesSegmented();
+    EXPECT_TRUE(result->HandleList.empty());
+    EXPECT_TRUE(result->NextMarker.empty());
+  }
+
+  TEST_F(FileShareFileClientTest, LeaseRelated)
+  {
+    std::string leaseId1 = CreateUniqueLeaseId();
+    auto aLease = *m_fileClient->AcquireLease(leaseId1);
+    EXPECT_FALSE(aLease.ETag.empty());
+    EXPECT_FALSE(aLease.LastModified.empty());
+    EXPECT_EQ(aLease.LeaseId, leaseId1);
+    aLease = *m_fileClient->AcquireLease(leaseId1);
+    EXPECT_FALSE(aLease.ETag.empty());
+    EXPECT_FALSE(aLease.LastModified.empty());
+    EXPECT_EQ(aLease.LeaseId, leaseId1);
+
+    auto properties = *m_fileClient->GetProperties();
+    EXPECT_EQ(properties.LeaseState.GetValue(), Files::Shares::LeaseStateType::Leased);
+    EXPECT_EQ(properties.LeaseStatus.GetValue(), Files::Shares::LeaseStatusType::Locked);
+
+    std::string leaseId2 = CreateUniqueLeaseId();
+    EXPECT_NE(leaseId1, leaseId2);
+    auto cLease = *m_fileClient->ChangeLease(leaseId1, leaseId2);
+    EXPECT_FALSE(cLease.ETag.empty());
+    EXPECT_FALSE(cLease.LastModified.empty());
+    EXPECT_EQ(cLease.LeaseId, leaseId2);
+
+    auto fileInfo = *m_fileClient->ReleaseLease(leaseId2);
+    EXPECT_FALSE(fileInfo.ETag.empty());
+    EXPECT_FALSE(fileInfo.LastModified.empty());
+
+    aLease = *m_fileClient->AcquireLease(CreateUniqueLeaseId());
+    properties = *m_fileClient->GetProperties();
+    auto brokenLease = *m_fileClient->BreakLease();
+    EXPECT_FALSE(brokenLease.ETag.empty());
+    EXPECT_FALSE(brokenLease.LastModified.empty());
+
+    aLease = *m_fileClient->AcquireLease(CreateUniqueLeaseId());
+    brokenLease = *m_fileClient->BreakLease();
+    EXPECT_FALSE(brokenLease.ETag.empty());
+    EXPECT_FALSE(brokenLease.LastModified.empty());
+    m_fileClient->BreakLease();
+  }
+
+  TEST_F(FileShareFileClientTest, ConcurrentUpload)
+  {
+    std::vector<uint8_t> fileContent = RandomBuffer(static_cast<std::size_t>(8_MB));
```

Comment:
i know you copied this from blob test case. But that one is so slow that i was gonna rewrite it.